Build a regression model.

In [2]:
import statsmodels.api as sm
import numpy as np
import pandas as pd
import math
from sklearn import linear_model


In [3]:
totalData = pd.read_csv('data/totalData.csv', sep=',')

In [4]:
#Getting rid of values that can't be used in regression such as strings and NULL's
X = totalData.drop(columns = ["names", "locations"])
for index, row in X.iterrows():
    for key in row.keys():
        if math.isnan(row[key]):
            try:
                X = X.drop(index)
            except:
                pass
#Making sure the data is the same shape            
y = X["free_bikes"]
X = X.drop(columns = ["free_bikes"])

X = sm.add_constant(X) 
lin_reg = sm.OLS(y,X.astype(float))

NameError: name 'math' is not defined

Provide model output and an interpretation of the results. 

In [21]:
model = lin_reg.fit()
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:             free_bikes   R-squared:                       0.113
Model:                            OLS   Adj. R-squared:                  0.109
Method:                 Least Squares   F-statistic:                     26.53
Date:                Mon, 20 Nov 2023   Prob (F-statistic):           5.99e-39
Time:                        10:21:19   Log-Likelihood:                -5357.7
No. Observations:                1674   AIC:                         1.073e+04
Df Residuals:                    1665   BIC:                         1.078e+04
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const          -1.24e+04    922.889    -13.436

We see here that we have an R-squared of only 0.113 for free bikes to all the data given. This shows that the independent variables are mostly not responsible for the variation in the free bikes axis.
There may be other factors outside of the data collected by FourSquare that determines how many bikes are free.

# Stretch

How can you turn the regression model into a classification model?